### ISSUE TITLE
git subrepo pull --all broken after 34a3a939

### ISSUE THREAD

**Issue by jrosdahl** [ingydotnet/git-subrepo#115]
As of 34a3a939, `--all` affects `merge-base`, which breaks `pull --all` since `pull` implicitly runs `merge-base`.

Here's what happened to me:

```
% git subrepo pull --all
Subrepo 'elisp/ext/dash' is up to date.
Subrepo 'elisp/ext/expand-region' is up to date.
Subrepo 'elisp/ext/flycheck' is up to date.
Subrepo 'elisp/ext/ido-vertical-mode' is up to date.
Subrepo 'elisp/ext/iflipb' is up to date.
Subrepo 'elisp/ext/multiple-cursors' is up to date.
Subrepo 'elisp/ext/phi-search' is up to date.
Valid: 71f0142edae6196535bfc27b79f317dc7a12ea1d = 7b851159e2914e0e1a82d0fd4ff429346a2b9def
Valid: 2100b89eb89a7e1e06b5a54ff52c6b9db1995208 = 772b9aa98eb678f9bcfadbe0727a68bef46fd3af
Valid: 1cf723a408800b489a5f74bd14219a74b665a7f3 = 8ea104a279c41f6be6c7d51fd9ae7f4c96c24f0c
Multiple on subrepo/elisp/ext/yasnippet: 9c813ff6082814a1c834658e627b2513614b514b = 8d8358d835a62e30265d5ec9caa66ed465e79b36
git-subrepo: Can't find a common ancestor between refs/subrepo/elisp/ext/yasnippet/fetch and subrepo/elisp/ext/yasnippet. This might be the case if someone used push --force on the subrepo. You need to find out why and possibly reclone the subrepo.
```

Pulling without `--all` worked fine:

```
% git subrepo pull elisp/ext/yasnippet
Subrepo 'elisp/ext/yasnippet' pulled from 'https://github.com/capitaomorte/yasnippet' (master).
```

Since I guess `merge-base` more or less only is a debugging tool, perhaps add a `merge-bases` subcommand that acts like "merge-base --all" if it's important to expose this functionality?

**Comment by grimmySwe** [comment]
Created a new branch issue/115 to see if I can fix this.

**Comment by grimmySwe** [comment]
Fixed this on issue/115 (49646494010ddbc5b392b7cf4a9549e029d3f659)

Merge base will now ignore --all for other commands.

**Comment by ingydotnet** [comment]
Fixed by #120
